### PR TITLE
fix: don't check subnets in vpc when rds subnet group provided

### PR DESCRIPTION
**Pull request by fnaranjo-vmw**
### Checklist:

* [ ] Have you added Release Notes in the docs repositories?
* [ ] Have you ran `make run-integration-tests` and `make run-terraform-tests`?
* [ ] Have you ran acceptance tests for the service under change?
* [x] Have you followed the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary)?

### CODE REVIEW

**Comment by blgm** [comment]
This is to address a bug that a user found. It seems that this validation logic isn't tested. While the change looks good, we missed an error in this code before, and I was thinking that some tests could help to guide us to the correct behaviour, and would also ensure that the behaviour doesn't get broken in the future. Maybe the right place for those tests would be here: https://github.com/cloudfoundry/csb-brokerpak-aws/blob/188beb8c5ab9ad35f290f7b95f2c74fbd26e55f3/terraform-tests/mssql_test.go#L283-L321

**Comment by fnaranjo-vmw** [comment]
> This is to address a bug that a user found. It seems that this validation logic isn't tested. While the change looks good, we missed an error in this code before, and I was thinking that some tests could help to guide us to the correct behaviour, and would also ensure that the behaviour doesn't get broken in the future. Maybe the right place for those tests would be here:
> 
> https://github.com/cloudfoundry/csb-brokerpak-aws/blob/188beb8c5ab9ad35f290f7b95f2c74fbd26e55f3/terraform-tests/mssql_test.go#L283-L321

Added tests in a new commit. New tests required some infra to be created (a VPC, some subnets and a rds subnet group).

**Comment by blgm** [comment]
This is good. Manually creating these resources is a trade-off that makes sense. In the future we could automate their creation using some AWS SDK calls. One tiny improvement that you're welcome to ignore would be to use constants at the top of the file, for example:
```go
const (
  // This resource needs to be manually created in the AWS console by ... 
  vpcWithMoreThan20Subnets = "vpc-066fae2eced5df2fb"

  // This resource needs to be manually created via the AWS CLI with the command ...
  securityGroupInVPCWithMoreThan20Subnets = "sg-in-vpc-more-than-20-subnets"
)
```

The idea being that it allows you to add a single comment that explains what the resource is and how to re-create it.

**Review by blgm — APPROVED** [review]

**Comment by fnaranjo-vmw** [comment]
> This is good. Manually creating these resources is a trade-off that makes sense. In the future we could automate their creation using some AWS SDK calls. One tiny improvement that you're welcome to ignore would be to use constants at the top of the file, for example:
> 
> ```go
> const (
>   // This resource needs to be manually created in the AWS console by ... 
>   vpcWithMoreThan20Subnets = "vpc-066fae2eced5df2fb"
> 
>   // This resource needs to be manually created via the AWS CLI with the command ...
>   securityGroupInVPCWithMoreThan20Subnets = "sg-in-vpc-more-than-20-subnets"
> )
> ```
> 
> The idea being that it allows you to add a single comment that explains what the resource is and how to re-create it that lives outside of the test. And if we did add AWS SDK calls to automatically create these resources, it would not require the test to change at all - which is nice decoupling.

Thanks for the suggestion!
Added in commit: c0dd23f3b5a8c2e24e0f64b1405025491d8b8c12

File: terraform-tests/mssql_test.go [hunk 1]
```diff
@@ -85,6 +85,11 @@ func testTerraformMssql(region string) {
 
 	validVPC := awsVPCID
 
+	// The following two resources were manually created
+	// in the AWS console to be able to test some corner-cases
+	vpcWithMoreThan20Subnets := "vpc-066fae2eced5df2fb"
```
Comment (blgm) [inline terraform-tests/mssql_test.go:90]: Can we make these `const`s? That way the compiler will fail if there's any attempt to modify them. I've seen some tricky test flakes that were causes by `var`s being accidentally modified in one test, causing subsequent tests to fail.

**Comment by blgm** [comment]
I tried running the tests and they didn't work. I haven't checked, but I think the reasons may be:
- the VPC name in GovCloud and us-west-2 is not the same
- the VPC and security group may have been removed by our cleanup scripts

**Comment by fnaranjo-vmw** [comment]
This change seems to have broken CI: https://runway-ci.eng.vmware.com/teams/service-enablement/pipelines/aws-csb/jobs/run_terraform_tests/builds/233
